components/sync15/src/sync_multiple.rs

```diff
@@ -60,8 +61,14 @@ pub fn sync_multiple(
     root_sync_key: &KeyBundle,
     sync_ping: &mut telemetry::SyncTelemetryPing,
     interruptee: &impl Interruptee,
+    service_status: &mut ServiceStatus,
 ) -> result::Result<HashMap<String, Error>, Error> {
```

Comment:
If we had this function return a `SyncResult`, instead of a `Result`, maybe that would avoid the outparam funkiness and error conversions...but I'm not sure we need the outparam at all, our errors seem rich enough where we could pass them up using `?`, and then `sync` would convert them to a `ServiceStatus` once. Does that make sense?